server/src/com/thoughtworks/go/server/service/EnvironmentConfigService.java

```diff
@@ -191,9 +191,29 @@ private void editEnvironments(EditEnvironments edit, EnvironmentConfig config, U
         }
     }
 
-    public List<EnvironmentPipelineModel> getAllPipelinesForUser(Username user) {
-        List<EnvironmentPipelineModel> pipelines = new ArrayList<>();
-        for (PipelineConfig pipelineConfig : goConfigService.getAllPipelineConfigs()) {
+    public List<EnvironmentPipelineModel> getAllLocalPipelinesForUser(Username user) {
+        List<PipelineConfig> pipelineConfigs = goConfigService.getAllLocalPipelineConfigs();
+        return getAllPipelinesForUser(user, pipelineConfigs);
+    }
+    public List<EnvironmentPipelineModel> getAllRemotePipelinesForUserInEnvironment(Username user,EnvironmentConfig environment) {
+        List<EnvironmentPipelineModel> pipelines = new ArrayList<EnvironmentPipelineModel>();
+        for (EnvironmentPipelineConfig pipelineConfig : environment.getRemotePipelines()) {
+            String pipelineName = CaseInsensitiveString.str(pipelineConfig.getName());
+            if (securityService.hasViewPermissionForPipeline(user, pipelineName)) {
+                if (environment != null) {
```

Comment:
Why do we need this null check here? If `environment` was `null`, wouldn't [this line ](https://github.com/gocd/gocd/pull/2265/files#diff-cf566ce8a9594182b030044a7a8398dbR200) throw an exception already?